### PR TITLE
[SHRINKRES-351] enh: Update components resulted from project-local resolution

### PR DESCRIPTION
**Pull request by lprimak**
Updates components (sisu, maven) where necessary
Adds Java 17 builds to CI matrix
~~Based, and will be rebased on #341 when it gets merged~~

closes #214

### CODE REVIEW

**Comment by lprimak** [comment]
TODO: successfully upgrade maven invoker component (not plugin)

**Comment by petrberan** [comment]
Thank you for looking into this @lprimak . I'll duplicate the main branch as new support and probably release Resolvers in version 4.x with JDK 17 support and leave JDK 8 for the 3.x branch

We should also update our test matrix for the JDK 17

**Comment by lprimak** [comment]
Figured out how to keep JDK 8 compatibility. Only checkstyle needed to be downgraded, which is no big deal.
Now compiles on 8,11 and 17 with the new features and component upgrades

**Comment by lprimak** [comment]
@petrberan Ready to go!

**Comment by lprimak** [comment]
@petrberan This one is ready to go as well now.
Thank you!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
         # https://api.adoptium.net/v3/info/available_releases
         # To list the available releases for a given "feature version" (example for 16):
         # https://api.adoptium.net/v3/assets/latest/16/hotspot
-        java: [8, 11]
+        java: [8, 11, 17]
```
Comment (petrberan) [inline .github/workflows/build.yml:31]: Can we have this as a separate commit please?

File: maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/logging/AetherLoggerFactory.java
Comment (petrberan) [inline maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/logging/AetherLoggerFactory.java:1]: Same here, separate commit please

File: maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java [hunk 2]
```diff
@@ -39,10 +39,10 @@ private void setOutputHandlers(String expectedRegex, CountDownLatch countDownLat
                 new ResolverErrorOutputHandler(logBuffer, expectedRegex, countDownLatch);
             ResolverOutputHandler outputHandler = new ResolverOutputHandler(logBuffer, expectedRegex, countDownLatch);
 
-            invoker.setOutputHandler(outputHandler);
+            invocationRequest.setOutputHandler(outputHandler);
```
Comment (petrberan) [inline maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java:42]: This seems to be duplicated

File: maven/impl-maven/pom.xml [hunk 3]
```diff
@@ -67,26 +67,22 @@
             <artifactId>plexus-classworlds</artifactId>
             <version>${version.org.codehaus.plexus-classworlds}</version>
         </dependency>
-        <dependency>
```
Comment (petrberan) [inline maven/impl-maven/pom.xml:70]: And same with plexus-component-annotations please

**Comment by petrberan** [comment]
Shame we can't keep the Guice on 7.0.0 though. Besides just some minor details this looks solid @lprimak , can you have a look at the comments above?

File: .github/workflows/build.yml (hunk 1 above)
Comment (lprimak) [reply .github/workflows/build.yml:31]: ok, will do

File: maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/logging/AetherLoggerFactory.java
Comment (lprimak) [reply maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/logging/AetherLoggerFactory.java:1]: Can't do that, as `AetherLoggerFactory.java` API dependencies are removed due to the upgrade

File: maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java (hunk 2 above)
Comment (lprimak) [reply maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java:42]: Hmmm... I don't see where / how

File: maven/impl-maven/pom.xml (hunk 3 above)
Comment (lprimak) [reply maven/impl-maven/pom.xml:70]: Unfortunately no, as it's deprecated / removed due to other upgrades

**Comment by lprimak** [comment]
Yup, Sisu is not compatible with Guice 7.
I will separate the CI Java 17 into separate commit.
I don't think the other comments are workable though.

Thanks for your help!

**Comment by lprimak** [comment]
Done!

File: maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java (hunk 2 above)
Comment (petrberan) [reply maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java:42]: L42 and L43 are the same - `invocationRequest.setOutputHandler(outputHandler);`
L45 and L46 are the same - `invocationRequest.setErrorHandler(errorOutputHandler);`

Or is it necessary to set the handlers 2 times?

File: maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/logging/AetherLoggerFactory.java
Comment (petrberan) [reply maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/logging/AetherLoggerFactory.java:1]: Got it, my bad

File: maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java (hunk 2 above)
Comment (lprimak) [reply maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/BuildTrigger.java:42]: Oh, I see it now. Thanks for pointing this out.
Fixed!

**Review by petrberan — APPROVED** [review]
LGTM

Thank you @lprimak , I'll merge this and release new Resolvers later this week

**Pull request merged**